### PR TITLE
[lldb][DWARFASTParserClang][NFCI] Extract DW_AT_data_member_location calculation logic

### PR DESCRIPTION
**Pull request by Michael137**
Currently this non-trivial calculation is repeated multiple times, making it hard to reason about when the `byte_offset`/`member_byte_offset` is being set or not.

This patch simply moves all those instances of the same calculation into a helper function.

We return an optional to remain an NFC patch. Default initializing the offset would make sense but requires further analysis and can be done in a follow-up patch.

### CODE REVIEW

**Comment by felipepiovezan** [comment]
image

**Comment by felipepiovezan** [comment]
this must have felt good to fix!

File: lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp [hunk 1]
```diff
@@ -519,6 +519,33 @@ TypeSP DWARFASTParserClang::ParseTypeFromDWARF(const SymbolContext &sc,
   return UpdateSymbolContextScopeForType(sc, die, type_sp);
 }
 
+static std::optional<uint32_t>
```
Comment (Michael137) [inline lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserClang.cpp:522]: The integer types are a bit inconsistent (but this didn't change with this patch).

* `ParseInheritance` stores the value in an `off_t`. Then passes it to `clang::CharUnits::fromQuantity` which takes an `int64_t`
* The Rust `Variant` support stores offsets as uint32_t, and so does `MemberAttributes`. But `FormValue::Unsigned()` returns a uint64_t.

Would be nice to make this consistent eventually

**Review by clayborg — APPROVED** [review]
Looks good!

**Pull request merged**